src/views/edition/components/sidebar.vue

```diff
@@ -111,67 +111,90 @@
 </template>
 
 <script lang="ts">
-import Vue from 'vue';
+import { Component, Prop, Emit, Vue } from 'vue-property-decorator';
 import { EditionInfo } from '@/models/edition';
 import EditionService from '@/services/edition';
 import { ImagedObject } from '@/models/imaged-object';
 import PermissionModal from './permission-modal.vue';
 import EditionIcons from '@/components/cues/edition-icons.vue';
 
-export default Vue.extend({
+
+@Component({
     name: 'edition-ver-sidebar',
     components: {
         PermissionModal,
-        EditionIcons,
-    },
-    props: {
-        page: String
-    },
-    data() {
-        return {
-            editionService: new EditionService(),
-            newCopyName: '',
-            waiting: false,
-            errorMessage: '',
-            newEditionName: '',
-            renaming: false
-        };
-    },
-    computed: {
-        readOnly(): boolean {
-            return this.current!.permission.readOnly;
-        },
-        isAdmin(): boolean {
-            return this.current!.permission.isAdmin;
-        },
-        canRename(): boolean {
-            return this.current!.permission.mayWrite;
-        },
-        user(): boolean {
+        EditionIcons
+    }
+})
+
+export default class SideBar extends Vue {
+
+    @Prop() protected page!: string;
+
+    // data => member parameters
+    // protected (not private) to allow future inheritance
+    // =======================================================
+
+    protected editionId: number = 0;
+    protected editionService: EditionService = new EditionService() ;
+    protected newCopyName: string = '';
+    protected waiting: boolean = false;
+    protected errorMessage: string =  '';
+    protected newEditionName: string =  '';
+    protected renaming: boolean = false;
+
+
+    // computed
+
+        public get currentEditionExists(): boolean {
+            // if ( undefined !== this.$state.editions.current ) {
+            //    this.waiting = false;
+            // }
+            return ( undefined !== this.$state.editions.current
+                     && null !== this.$state.editions.current) ;
+        }
+
+        public get currentEdition(): EditionInfo {
+            return this.$state.editions.current! || null; // {};
```

Comment:
This makes no sense and should result in a compilation error, since you're returning EditionInfo | null and the function is supposed to return EditionInfo. currentEdition should be this.$state.editions.current, as mentioned before.